src/models.jl

```diff
@@ -0,0 +1,156 @@
+"""
+    Models for random hypergraphs building:
+        * random model
+        * k-uniform model
+        * d-uniform model
+        * preferential-attachment
+"""
+
+
+"""
+    random_model(nVertices, nEdges)
+
+Generate a *random* hypergraph without any structural constraints.
+
+**The Algorithm**
+Given two integer parameters *nVertices* and *nEdges* (the number of nodes and hyperedges, respectively),
+the algorithm computes - for each hyperedge *he={1,...,m}* -
+a random number *s ϵ [1, n]* (i.e. the hyperedge size).
+Then, the algorithm selects uniformly at random *s* vertices from *V* to be added in *he*.
+"""
+function random_model(nVertices::Int, nEdges::Int)
+    mx = Matrix{Union{Nothing,Bool}}(nothing, nVertices,nEdges)
+    for e in 1:size(mx,2)
+        nv = rand(1:size(mx,1))
+        mx[sample(1:size(mx,1), nv;replace=false), e] .= true
+    end
+
+    h = Hypergraph(mx)
+    if all(length.(h.v2he) .> 0)
+        return h
+    else
+        return random_model(nVertices, nEdges)
+    end
+end
+
+
+"""
+    random_kuniform_model(nVertices, nEdges, k)
+
+Generates a *k*-uniform hypergraph, i.e. an hypergraph where each hyperedge has size *k*.
+
+**The Algorithm**
+The algorithm proceeds as the *randomH*, forcing the size of each hyperedge equal to *k*.
+"""
+function random_kuniform_model(nVertices::Int, nEdges::Int, k::Int)
+    mx = Matrix{Union{Nothing,Bool}}(nothing, nVertices,nEdges)
+    for e in 1:size(mx,2)
+        nv = k
+        mx[sample(1:size(mx,1), nv;replace=false), e] .= true
+    end
+
+    h = Hypergraph(mx)
+    if all(length.(h.he2v) .== k)
```

Comment:
here it should probably an assertion - right?

Reply:
We use it as control, to generate again the random graph, not to assert an error

Reply:
in what case it is possible that the edge does not have `k` vertices in this algorithm?

Reply:
Yes, you are right. COVID effect on myself. I changed